crypto/keys/src/traits.rs

```diff
@@ -106,10 +123,15 @@ pub trait PublicKey:
     + Ord
     + Serialize
     + Sized
-    + for<'bytes> TryFrom<&'bytes [u8]>
+    + for<'bytes> TryFrom<&'bytes [u8], Error = KeyError>
```

Comment:
this change (adding `Error = KeyError` bound) makes it much easier to implement the generic cryptobox, otherwise I have to assume that any error type could fly out of this thing. the only thing not returning `KeyError` was `ed25519` which was returning a default `SignatureError`, so seems like not a big deal.